diff --git a/KernelSU/kernel/ksud.c b/KernelSU/kernel/ksud.c
index 0000000..1111111 100644
--- a/KernelSU/kernel/ksud.c
+++ b/KernelSU/kernel/ksud.c
@@ -487,7 +487,7 @@ void ksu_handle_sys_read(unsigned int fd)
 
-    if (!d_is_reg(fp->f_path.dentry)) {
+    if (!d_is_reg(file->f_path.dentry)) {
         goto out;
     }
 
@@ -494,7 +494,7 @@ void ksu_handle_sys_read(unsigned int fd)
-    const char *short_name = fp->f_path.dentry->d_name.name;
+    const char *short_name = file->f_path.dentry->d_name.name;
 
     if (strcmp(short_name, "init.rc")) {
         goto out;
@@ -502,7 +502,7 @@ void ksu_handle_sys_read(unsigned int fd)
     char path[256];
-    char *dpath = d_path(&fp->f_path, path, sizeof(path));
+    char *dpath = d_path(&file->f_path, path, sizeof(path));
 
     if (IS_ERR(dpath)) {
         goto out;
